cmd/root.go

```diff
@@ -356,7 +356,7 @@ func ExecuteContext(ctx context.Context, rootCmd *cobra.Command, clients *shared
 				clients.IO.PrintDebug(ctx, "... root cleanup waitgroup done.")
 				cleanup(ctx, clients)
 				clients.IO.PrintDebug(ctx, "Exiting with cancel exit code.")
-				os.Exit(int(iostreams.ExitCancel))
+				clients.Os.Exit(int(iostreams.ExitCancel))
```

Comment:
note: Switched a few of these `os.Exit` to our mock-able `clients.Os.Exit`. This is necessary for the tests to pass, otherwise the entire process is terminated.